src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.hpp

```diff
@@ -368,18 +370,40 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_AttachToThreadN
 }
 
 template <class ImplClass>
-void GenericThreadStackManagerImpl_OpenThread<ImplClass>::_OnThreadAttachFinished()
+void GenericThreadStackManagerImpl_OpenThread<ImplClass>::CancelOnGoingOperations(void)
 {
     if (mpConnectCallback != nullptr)
     {
-        DeviceLayer::SystemLayer().ScheduleLambda([this]() {
-            VerifyOrReturn(mpConnectCallback != nullptr);
-            mpConnectCallback->OnResult(NetworkCommissioning::Status::kSuccess, CharSpan(), 0);
-            mpConnectCallback = nullptr;
-        });
+        Impl()->SetThreadEnabled(false);
+        OnAttachEnd(NetworkCommissioning::Status::kUnknown);
```

Comment:
Suggest `kOtherConnectionFailure` instead --> not unknown, it's more that we have not yet attached. Too bad we don't have a better value though...